test/eventbus.test.ts

```diff
@@ -0,0 +1,114 @@
+import { aws_events, aws_lambda, Stack } from "aws-cdk-lib";
+import { EventBus, EventBusRule } from "../src/event-bridge";
+import { EventBusTransform } from "../src/event-bridge/transform";
+import { Function } from "../src/function";
+
+let stack: Stack;
+
+beforeEach(() => {
+  stack = new Stack();
+});
+
+test("new bus from aws bus", () => {
+  const bus = new aws_events.EventBus(stack, "bus");
+
+  EventBus.fromBus(bus);
+});
+
+test("new bus without wrapper", () => {
+  new EventBus(stack, "bus");
+});
+
+test("new rule without when", () => {
+  const bus = new EventBus(stack, "bus");
+
+  const rule = new EventBusRule(stack, "rule", bus, (_event) => true);
+
+  expect(rule.rule._renderEventPattern()).toEqual({ source: [{ prefix: "" }] });
+});
+
+test("new transform without map", () => {
+  const bus = new EventBus(stack, "bus");
+
+  const rule = new EventBusRule(stack, "rule", bus, (_event) => true);
+  const transform = new EventBusTransform((event) => event.source, rule);
+
+  expect(transform.targetInput.bind(rule.rule)).toEqual({
+    inputPath: "$.source",
+  } as aws_events.RuleTargetInputProperties);
+});
+
+test("rule from existing rule", () => {
+  const awsRule = new aws_events.Rule(stack, "rule");
+
+  const rule = EventBusRule.fromRule(awsRule);
+  const transform = new EventBusTransform((event) => event.source, rule);
+
+  expect(transform.targetInput.bind(rule.rule)).toEqual({
+    inputPath: "$.source",
+  } as aws_events.RuleTargetInputProperties);
+});
+
+test("new bus with when", () => {
+  const rule = new EventBus(stack, "bus").when(stack, "rule", () => true);
+
+  expect(rule.rule._renderEventPattern()).toEqual({ source: [{ prefix: "" }] });
+});
+
+test("new bus with when pipe event bus", () => {
+  const busBus = new EventBus(stack, "bus");
+
+  const rule = busBus.when(stack, "rule", () => true);
+  rule.pipe(busBus);
+
+  expect((rule.rule as any).targets.length).toEqual(1);
+  expect(
+    (rule.rule as any).targets[0] as aws_events.IRuleTarget
+  ).toHaveProperty("arn");
+});
+
+test("new bus with when map pipe function aa", () => {
```

Comment:
`aa` ?

Reply:
ha... my lazy way to break the compile cache was to add letters to the test names. I missed one or two I guess.